src/db/VectorDB.cpp

```diff
@@ -0,0 +1,77 @@
+#include <mw/db/VectorDB.h>
+#include "common/Database.h"
+#include "common/Serializable.h"
+
+VectorDB::VectorDB(const std::string& name, libmw::IDBWrapper* pDBWrapper, libmw::IDBBatch* pBatch)
+    : m_name("V" + name), m_pDatabase(std::make_unique<Database>(pDBWrapper, pBatch))
+{
+}
+
+VectorDB::~VectorDB() {}
+
+uint64_t VectorDB::Size() const
```

Comment:
I don't think the maintenance headaches caused by this size field are actually necessary. We should be able to calculate size easily by simply looking at how many hashes are in the hash_pmmr. You should be able to calculate size using: `Index::At(m_pHashFile->GetSize() / mw::Hash::size()).GetLeafIndex()`